### PR TITLE
update various dependencies

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: spinnaker-dependencies/spinnaker-dependencies.gradle [hunk 1]
```diff
@@ -40,14 +40,14 @@ dependencies {
   // this project and need to configure gradle plugins etc.
   api(platform("org.jetbrains.kotlin:kotlin-bom:$kotlinVersion"))
   api(platform("org.junit:junit-bom:5.6.2"))
-  api(platform("com.fasterxml.jackson:jackson-bom:2.10.3"))
+  api(platform("com.fasterxml.jackson:jackson-bom:2.10.4"))
```
Comment (robfletcher) [inline spinnaker-dependencies/spinnaker-dependencies.gradle:43]: This is in-line with the version of Spring in #663

**Comment by cfieber** [comment]
have you tried starting up an app with this change in it?

**Comment by robzienert** [comment]
Please also test OSS and -nflx paths.

**Comment by robfletcher** [comment]
I've backed out the change to Jackson as we ended up backing down a couple of Spring patch versions. That means all the updates here, with the exception of `mysql-connector-java` are test related.

I have tested this against keel and keel-nflx.

In the long term I think there are some things (Jackson and mysql-connector-java among them) that we could and probably should derive from `spring-boot-dependencies.bom`.

**Pull request merged**